pkg/translator/ingress_translator.go

```diff
@@ -363,52 +367,49 @@ func (it *IngressTranslator) updateEdgeLBPoolObject(pool *models.V2Pool, backend
 		}
 	}
 
-	// frontendExists holds whether the EdgeLB frontend that corresponds to the current Ingress resource exists in the EdgeLB pool.
-	// It is used to understand whether we need to create it or not after we have iterated all EdgeLB frontends present in the EdgeLB pool.
-	frontendExists := false
-	// updatedFrontends holds the set of updated EdgeLB frontends.
-	// It is used as the final set of EdgeLB frontends for the EdgeLB pool if we find out we need to update it.
-	updatedFrontends := make([]*models.V2Frontend, 0, len(pool.Haproxy.Frontends))
-
-	// Iterate over the EdgeLB pool's frontends and check whether each frontend is owned by the current Ingress.
-	// In case a frontend isn't owned by the current Ingress, it is left unchanged and added to the set of "updated" frontends.
-	// Otherwise, it is checked for correctness and, if necessary, replaced with the computed frontend for the current Ingress.
-	for _, frontend := range pool.Haproxy.Frontends {
-		// Parse the name of the EdgeLB frontend in order to determine if the current Ingress owns it.
-		// If the current EdgeLB frontend isn't owned by the current Ingress, it is left unchanged.
-		frontendMetadata, err := computeIngressOwnedEdgeLBObjectMetadata(frontend.Name)
-		if err != nil || !frontendMetadata.IsOwnedBy(it.ingress) {
-			updatedFrontends = append(updatedFrontends, frontend)
-			report.Report("no changes required for frontend %q (not owned by %s)", frontend.Name, kubernetesutil.Key(it.ingress))
-			continue
-		}
-		// At this point we know the current EdgeLB frontend is owned by the current Ingress.
-		// Check whether the Ingress resource has been deleted, and skip (i.e. remove) the EdgeLB frontend if it has.
-		if ingressDeleted {
-			wasChanged = true
-			report.Report("must delete frontend %q as %q was deleted", frontend.Name, kubernetesutil.Key(it.ingress))
-			continue
+	// Check if pool's frontends match the desired list. If a frontend is not
+	// managed by dklb it's added to the list of desiredFrontends.
+	desiredFrontends := computeEdgeLBFrontendForIngress(it.ingress, *it.spec)
+	updatedFrontends := make([]*models.V2Frontend, 0)
+	if !reflect.DeepEqual(desiredFrontends, pool.Haproxy.Frontends) {
+		wasChanged = true
+		report.Report("frontend list requires an update")
+		for _, frontend := range pool.Haproxy.Frontends {
+			frontendMetadata, err := computeIngressOwnedEdgeLBObjectMetadata(frontend.Name)
+			if err != nil || !frontendMetadata.IsOwnedBy(it.ingress) {
```

Comment:
if `err != nil` we don't log anything. Perhaps it won't be bad to log that error.

Reply:
It'll only add noise in the logs. When the function returns an error it means the frontend needs to be added to the list because it's not managed by dklb. The code could have been written better initially though.